code/modules/admin/verbs/playsound.dm

```diff
@@ -44,9 +44,22 @@ GLOBAL_LIST_EMPTY(sounds_cache)
 	set name = "Play Local Sound"
 	if(!check_rights(R_SOUNDS))	return
 
-	log_admin("[key_name(src)] played a local sound [S]")
-	message_admins("[key_name_admin(src)] played a local sound [S]", 1)
-	playsound(get_turf(src.mob), S, 50, 0, 0)
+	var/sound/uploaded_sound = sound(S, channel = CHANNEL_ADMIN)
+	uploaded_sound.priority = 250
+
+	GLOB.sounds_cache += S
+
+	if(alert("Are you sure?\nSong: [S]\nNow you can also play this sound using \"Play Local Sound\".", "Confirmation request", "Play", "Cancel") == "Cancel")
+		return
+
+	log_admin("[key_name(src)] played local sound [S]")
+	message_admins("[key_name_admin(src)] played local sound [S]")
+
+	for(var/mob/M in GLOB.player_list)
+		if(M.client.prefs.sound & SOUND_MIDI)
+			uploaded_sound.volume = 100 * M.client.prefs.get_channel_volume(CHANNEL_ADMIN)
+			SEND_SOUND(get_turf(src.mob), uploaded_sound)
```

Comment:
Again. This is playing the same sound over and over. You need to send this to the mobs themselves in range.

Or, alternatively, just play the sound down the `CHANNEL_ADMIN` channel.

If I have to make a **fourth** review saying `test your code` I am going to lose it.